src/components/datacollection/DataPageMenu.svelte

```diff
@@ -6,68 +6,51 @@
 
 <script lang="ts">
   import { t } from '../../i18n';
-  import { createMenu } from 'svelte-headlessui';
   import ClearIcon from 'virtual:icons/ri/delete-bin-2-line';
   import UploadIcon from 'virtual:icons/ri/upload-2-line';
   import DownloadIcon from 'virtual:icons/ri/download-2-line';
   import MenuItems from '../control-bar/control-bar-items/MenuItems.svelte';
   import MenuItem from '../control-bar/control-bar-items/MenuItem.svelte';
-  import MenuTransition from '../MenuTransition.svelte';
   import IconButton from '../IconButton.svelte';
   import MoreIcon from 'virtual:icons/mdi/dots-vertical';
+  import { createDropdownMenu } from '@melt-ui/svelte';
 
   export let downloadDisabled = false;
   export let clearDisabled = false;
   export let onClearGestures: () => void;
   export let onDownloadGestures: () => void;
   export let onUploadGestures: () => void;
 
-  const menu = createMenu({ label: $t('content.data.controlbar.button.menu') });
-
-  const onSelect = (event: Event) => {
-    const { selected } = (event as CustomEvent).detail;
-    switch (selected) {
-      case 'upload': {
-        onUploadGestures();
-        break;
-      }
-      case 'download': {
-        onDownloadGestures();
-        break;
-      }
-      case 'clear': {
-        onClearGestures();
-        break;
-      }
-    }
-    menu.set({ selected: null });
-  };
+  const menu = createDropdownMenu({ forceVisible: true });
+  const { trigger } = menu.elements;
+  const { open } = menu.states;
 </script>
 
 <div class="relative inline-block leading-none">
   <IconButton
     ariaLabel={$t('content.data.controlbar.button.menu')}
     rounded
-    useAction={menu.button}
-    on:select={onSelect}>
+    {...$trigger}
+    useAction={trigger}>
```

Comment:
Not sure how to pass preprocessor form to `IconButton` so have left it as the longer form when using the `IconButton` component

Reply:
I think the settings and help menu buttons should also be IconButtons and use the same syntax.

Reply:
Took me a while to figure out how to focus on a component, but done here
https://github.com/microbit-foundation/ml-trainer/pull/171/commits/9dacf38f5e06d9f413e6c4665cbbb238f95db531 and here https://github.com/microbit-foundation/ml-trainer/pull/171/commits/b3629f0a3a0a37663477074216f8eed1e0e86675